### ISSUE TITLE
a minus seems to influence all of the terms, not only the one next to it

### ISSUE THREAD

**Issue by va1en0k** [rm-hull/infix#24]
(Tested in both clojure and clojurescript)

    (infix.macros/infix 1 - 1 + 1)
    => -1

    (infix.macros/infix 1 - 2 + 3)
    => -4

I'd expect the values to be `1` and `2`.

And thanks for the library, it's quite useful!

**Comment by rm-hull** [comment]
Sounds like it is related to #4 - will investigate

**Comment by rm-hull** [comment]
Added some tests in commit 0be9d32 for both the `infix` macro and the `from-string` for the cases you documented above - and interestingly, only the `infix` tests fail. See https://travis-ci.org/rm-hull/infix/builds/245542516#L334-L340

**Comment by rm-hull** [comment]
Using macroexpand to try and debug whats going on:

```clojure
(macroexpand-1 '(infix 1 - 1 + 1))
; => (- 1 (+ 1 1))

(macroexpand-1 '(infix 1 - 2 + 3))
; => (- 1 (+ 2 3))
```

Whereas I think it should be evaluating as `(+ (- 1 1) 1)` and `(+ (- 1 2) 3)` respectively.

**Comment by va1en0k** [comment]
Ah, so it doesn't recognize the `-` and thus https://github.com/rm-hull/infix/blob/master/src/infix/core.clj#L131 fails to find it?

**Comment by rm-hull** [comment]
The fix (see commit c7e1d58) looks to be fairly simple - and all the existing tests pass, as do the new tests: https://travis-ci.org/rm-hull/infix/builds/245551709.

I might see if there is a comprehensive suite of arithmetic operator precedence tests I can port into test cases, as clearly there are some gaps here.. if you know of anything (or can contribute such) please let me know.

I'll release this as is with a version patch bump.
Thanks for the bug report though :)

**Comment by rm-hull** [comment]
0.2.12 is published to clojars
